test/modern_bpf/test_suites/syscall_exit_suite/accept4_x.cpp

```diff
@@ -1,11 +1,233 @@
+	assert_syscall_state(SYSCALL_SUCCESS, "accept4 (server)", connected_socket_fd, NOT_EQUAL, -1);
+
+	/* Cleaning phase */
+	syscall(__NR_shutdown, connected_socket_fd, 2);
+	syscall(__NR_shutdown, server_socket_fd, 2);
+	syscall(__NR_shutdown, client_socket_fd, 2);
+	syscall(__NR_close, connected_socket_fd);
+	syscall(__NR_close, server_socket_fd);
+	syscall(__NR_close, client_socket_fd);
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: fd (type: PT_FD) */
+	evt_test->assert_numeric_param(1, (int64_t)connected_socket_fd);
+
+	/* Parameter 2: tuple (type: PT_SOCKTUPLE) */
+	/* The server performs an `accept` so the `client` is the src. */
+	evt_test->assert_tuple_inet6_param(2, PPM_AF_INET6, IPV6_CLIENT, IPV6_SERVER, IPV6_PORT_CLIENT_STRING, IPV6_PORT_SERVER_STRING);
+
+	/* Parameter 3: queuepct (type: PT_UINT8) */
+	/* we expect 0 elements in the queue so 0%. */
+	evt_test->assert_numeric_param(3, (uint8_t)0);
+
+	/* Parameter 4: queuelen (type: PT_UINT32) */
+	/* we expect 0 elements. */
+	evt_test->assert_numeric_param(4, (uint32_t)0);
+
+	/* Parameter 5: queuemax (type: PT_UINT32) */
+	evt_test->assert_numeric_param(5, (uint32_t)QUEUE_LENGTH);
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	evt_test->assert_num_params_pushed(5);
+}
+
+#ifdef __NR_unlinkat
+TEST(SyscallExit, accept4X_UNIX)
+{
+	auto evt_test = new event_test(__NR_accept4, EXIT_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	int32_t client_socket_fd = 0;
+	int32_t server_socket_fd = 0;
+	struct sockaddr_un client_addr = {0};
+	struct sockaddr_un server_addr = {0};
+	evt_test->connect_unix_client_to_server(&client_socket_fd, &client_addr, &server_socket_fd, &server_addr);
+
+	/* We don't want to get any info about the connected socket so `addr` and `addrlen` are NULL. */
+	struct sockaddr *addr = NULL;
+	socklen_t *addrlen = NULL;
+	int flags = 0;
+	int connected_socket_fd = syscall(__NR_accept4, server_socket_fd, addr, addrlen, flags);
+	assert_syscall_state(SYSCALL_SUCCESS, "accept4 (server)", connected_socket_fd, NOT_EQUAL, -1);
+
+	/* In unix sockets the maximum queue length seems to be 512. */
+	FILE *f = fopen("/proc/sys/net/unix/max_dgram_qlen", "r");
+	if(f == NULL)
+	{
+		FAIL() << "'fopen' must not fail." << std::endl;
+	}
+	int unix_max_queue_len = 0;
+	if(fscanf(f, "%d", &unix_max_queue_len) != 1)
+	{
+		fclose(f);
+		FAIL() << "'fscanf' must not fail." << std::endl;
+	}
+	fclose(f);
+
+	/* Cleaning phase */
+	syscall(__NR_shutdown, connected_socket_fd, 2);
```

Comment:
Perhaps, we can also create a small helper to clean up the established stuff from `connect_x_client_to_server`

Reply:
Yeah, it sounds good, we have to pass all the sockets `fds`, but in the end, probably it would be clearer :)

Reply:
well, another idea is to encapsulate the fds in a struct, similar as for the `proc_info` struct for the process family calls-

Reply:
That seems a smart idea, I will give it a try thank you!!